solr/core/src/java/org/apache/solr/util/RedactionUtils.java

```diff
@@ -21,7 +21,7 @@
 
 public class RedactionUtils {
   public static final String SOLR_REDACTION_SYSTEM_PATTERN_PROP = "solr.redaction.system.pattern";
-  private static Pattern pattern = Pattern.compile(System.getProperty(SOLR_REDACTION_SYSTEM_PATTERN_PROP, ".*password.*"), Pattern.CASE_INSENSITIVE);
+  private static Pattern pattern = Pattern.compile(System.getProperty(SOLR_REDACTION_SYSTEM_PATTERN_PROP, ".*(password|secret|basicauth).*"), Pattern.CASE_INSENSITIVE);
```

Comment:
https://solr.apache.org/security.html#cve-2023-50291-apache-solr-can-leak-certain-passwords-due-to-system-property-redaction-logic-inconsistencies

> Users who cannot upgrade can also use the following Java system property to fix the issue:
> -Dsolr.redaction.system.pattern=".*(password|secret|basicauth).*"